service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -117,9 +119,8 @@ private void insertAssessmentResult(
   @Override
   public void setOffRampReason(Claim claimWithOffRamp) {
     List<ClaimSubmissionEntity> claimSubmissionList =
-        claimSubmissionRepository.findByReferenceIdAndIdType(
-            String.valueOf(claimWithOffRamp.getCollectionId()), DEFAULT_ID_TYPE);
-    Collections.reverse(claimSubmissionList);
+        claimSubmissionRepository.findByReferenceIdAndIdTypeOrderByCreatedAtDesc(
+            claimWithOffRamp.getCollectionId(), claimWithOffRamp.getIdType());
     ClaimSubmissionEntity claimSubmissionEntity = claimSubmissionList.get(0);
```

Comment:
@ianwolson , did you look at this via debug/breakpoints and verified if it's actually returning an ordered ResultSet as we expect?